zenoh-flow/src/io/input.rs

```diff
@@ -75,13 +76,17 @@ impl Inputs {
     ///
     /// If the underlying data is not relevant then these additional operations can be avoided by
     /// calling `take_raw` and using an [`InputRaw`](`InputRaw`) instead.
-    pub fn take<T: ZFData>(&mut self, port_id: impl AsRef<str>) -> Option<Input<T>> {
+    pub fn take<T: Send + Sync + 'static>(
+        &mut self,
+        port_id: impl AsRef<str>,
+        deserializer: impl Fn(&[u8]) -> anyhow::Result<T> + Send + Sync + 'static,
```

Comment:
Why this is not `deserializer: DeserializeFn<T>` ?
with `type DeserializerFn<T> =  Fn(&[u8]) -> anyhow::Result<T> + Send + Sync;`

Reply:
My main motivation is to make the signature of the method clearer to users: with `impl Fn(&[u8]) -> …` they know what is expected, if it's "hidden" behind `DeserializerFn` then they need to look that up first.

Plus, as `Fn` is a trait we need to put `deserializer: &DeserializerFn<T>` (or a box or an Arc) which feels less straightforward than being able to put anything that implements the trait.

I find these arguments compelling enough to keep it as is but I did check and putting `deserializer: &DeserializerFn<T>` would also work